### ISSUE TITLE
[Code Health] Make `#ConsensusVersion()` configurable

### ISSUE THREAD

**Issue by bryanchriswhite** [pokt-network/poktroll#395]
## Objective

Support compile-time configuration of the return of modules' `#ConsensusVersion()` methods.

## Origin Document

![image](https://github.com/pokt-network/poktroll/assets/600733/3e046eac-2785-451f-8211-4a63d17deadf)

## Goals

- Forward compatibility with alternative consensus versions without requiring code changes.

## Deliverables

- [x] BEFORE implementing any business logic PR: investigate what the plans for this are in the Cosmos SDK.
	- Understand what/why are their patterns
	- Compare it to `pocket-v0` and other blockchains
	- Etc...
- [x] A PR that factors out the magic numbers from all modules' `#ConsensusVersion()` methods to reference the same variable, whose default value is `1` but is configurable at compile-time via `ldflags -X ...` (see: [go link --help](https://pkg.go.dev/cmd/link)).

## Non-goals / Non-deliverables

- Any new business logic
- Contributing back to the Cosmos SDK

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @bryanchriswhite 
**Co-Owners**: @Olshansk

**Comment by Olshansk** [comment]
@bryanchriswhite Added some small extra details AND moved this to the `Shannon MainNet` milestone.

**Comment by Olshansk** [comment]
@okdas Can you confirm if you resolved this with the upgrade mechanism?

**Comment by okdas** [comment]
> @okdas Can you confirm if you resolved this with the upgrade mechanism?

It should be, but I'll check if `ldflags` is something we should consider using here.

**Comment by okdas** [comment]
TLDR: we don't need this.


Per https://docs.cosmos.network/main/build/building-modules/upgrade#consensus-version:

> # Consensus Version
> 
> Successful upgrades of existing modules require each AppModule to implement the function ConsensusVersion() uint64.
> 
> - The versions must be hard-coded by the module developer.
> - The initial version must be set to 1.
> 
> Consensus versions serve as state-breaking versions of app modules and must be incremented when the module introduces breaking changes.

Now, I have not tested and replicated a consensus version upgrade/migration of the module - but we are going to start experimenting with them on Beta testnets.
